sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveExternalCatalog.scala

```diff
@@ -523,18 +523,29 @@ private[spark] class HiveExternalCatalog(conf: SparkConf, hadoopConf: Configurat
           statsProperties += (columnStatKeyPropName(colName, k) -> v)
         }
       }
-      tableDefinition.copy(properties = tableDefinition.properties ++ statsProperties)
+      newTableDefinition.copy(properties = newTableDefinition.properties ++ statsProperties)
     } else {
-      tableDefinition
+      newTableDefinition
     }
 
-    if (tableDefinition.tableType == VIEW) {
-      client.alterTable(withStatsProps)
+    if (newTableDefinition.tableType == VIEW) {
+      client.alterTable(maybeWithStatsPropsTable)
     } else {
-      val oldTableDef = getRawTable(db, withStatsProps.identifier.table)
```

Comment:
To the other reviewers: `oldTableDef ` actually is storing the raw table metadata. In the new changes, it is renamed to `oldRawTableDef `